eval/evaluator.py

```diff
@@ -28,6 +34,47 @@ def __init__(self, config, analyzer=None, metric_handler=None, label_schema=None
+            },
+            {
+                "metric_name": "response_quality_score", 
+                "metric_desc": "Score for response quality", 
+                "metric_prompt": "Establish a revised five-point assessment scale for evaluating the quality of the response to a given prompt that emphasizes precise differentiation between all levels of accuracy and relevance. Assign the classification labels '1', '2', '3', '4', and '5', with each level distinctly representing the depth and accuracy of the response: 1. Deficient Accuracy and Lacking Relevance ('1'): Responses that fundamentally misinterpret or ignore the prompt and display a lack of relevance, signaling a complete disconnect and an ineffective response. 2. Basic Accuracy and Low Relevance ('2'): Responses that mention aspects of the prompt without depth and show only a low level of relevance, resulting in a response that lacks persuasive power and engagement. 3. Adequate Accuracy and Moderate Relevance ('3'): Responses that reflect the prompt accurately and convey moderate relevance, offering a fair and constructive response that is neither overly detailed nor overly superficial. 4. Detailed Accuracy and High Relevance ('4'): Responses that capture the nuances of the prompt in detail and exhibit high relevance, presenting a response that is both engaging and substantively insightful. 5. Exceptional Accuracy and Complete Relevance ('5'): Responses that deeply engage with the prompt, demonstrating both an exceptional level of accuracy and complete relevance, significantly enhancing the analysis and effectively addressing the prompt. This adjusted scale is designed to correct prior misclassifications by ensuring clear and accurate recognition of each level of response quality, thereby avoiding the underestimation of high-quality responses and providing a fair representation for lower-quality outputs."
+            },
+        ]
+    
+    def build_score_function(self, input_prompt, prompt_file, config: dict):
```

Comment:
The function should be used in the `get_eval_function` 
This will enrich the score dictionary with the implementation of the score function

Reply:
Should I put this function into the `get_eval_function`? Or should I put this function in the `eval_utils.py`?

Reply:
The function location is fine. Currently, you are not using this function.
In `the get_eval_function `  you get back a list of metrics (with their description and prompt). 
Now we would like to convert these prompts to score functions using this `build_score_function`
In the end the result should be a list of metrics with their name, description and their **score functions**

Reply:
Thanks for moving my function to `metric_gen.py`